typings/primordials.d.ts

```diff
@@ -343,7 +343,6 @@ declare namespace primordials {
   export import RegExp = globalThis.RegExp;
   export const RegExpPrototype: typeof RegExp.prototype
   export const RegExpPrototypeExec: UncurryThis<typeof RegExp.prototype.exec>
-  export const RegExpPrototypeCompile: UncurryThis<typeof RegExp.prototype.compile>
```

Comment:
Let's instead mark them as deprecated, keeping the types correct.
```suggestion
  /** @deprecated **/
  export const RegExpPrototypeCompile: UncurryThis<typeof RegExp.prototype.compile>
```

Reply:
I understand your concerns. I have a few questions based on your feedback:

1. Does Node.js have a process for reviewing and removing deprecated methods in major release cycles? 
2. Your idea to mark these methods as deprecated is indeed practical, but eventually, deprecated methods need to be removed. Wouldn’t it be more effective to implement lint rules in the primordials.d.ts file to prevent the addition of deprecated methods?

I would appreciate your thoughts on this.

Reply:
> Does Node.js have a process for reviewing and removing deprecated methods in major release cycles?

Node.js has a deprecation cycle that's defined in https://github.com/nodejs/node/blob/27f1306ee17b32db30ba9949d374482eb3b2c74c/doc/contributing/collaborator-guide.md#L487-L495
However, the methods you list in the OP do not apply, because:
- they are not Node.js APIs, they are defined by the ECMAScript spec and implemented by V8.
- they are not deprecated by us.
- the file you're modifying is only used for internal development, and deprecation only apply to public APIs.

> eventually, deprecated methods need to be removed

I don't think that's correct, a deprecated feature can stay around forever.



> Wouldn’t it be more effective to implement lint rules in the `primordials.d.ts` file to prevent the addition of deprecated methods?

I very much doubt so, regardless if those methods are defined in `primordials.d.ts`, they exist, and therefor can be used in `.js` files. If you do add `@deprecated` flag to them, IMO that would be more effective because folks who use an IDE with TS support would see those methods as underlined if they try to use it – if you remove it, it would only remove the autocompletion and the documentation.

Reply:
Another option is to actually delete these methods (or not add them in the first place) from the `primordials` object.